src/encoder/plugins/OpusEncoderPlugin.cxx

```diff
@@ -52,30 +53,33 @@ class OpusEncoder final : public OggEncoder {
 
 	ogg_int64_t packetno = 0;
 
-	ogg_int64_t granulepos;
+	ogg_int64_t granulepos = 0;
 
 public:
-	OpusEncoder(AudioFormat &_audio_format, ::OpusEncoder *_enc);
+	OpusEncoder(AudioFormat &_audio_format, ::OpusEncoder *_enc, bool _chaining);
 	~OpusEncoder() override;
 
 	/* virtual methods from class Encoder */
 	void End() override;
 	void Write(const void *data, size_t length) override;
 
-	size_t Read(void *dest, size_t length) override;
+	void PreTag() override;
+	void SendTag(const Tag &tag) override;
 
 private:
 	void DoEncode(bool eos);
 	void WriteSilence(unsigned fill_frames);
 
+	void GenerateHeaders(const Tag *tag);
 	void GenerateHead();
-	void GenerateTags();
+	void GenerateTags(const Tag *tag);
 };
 
 class PreparedOpusEncoder final : public PreparedEncoder {
 	opus_int32 bitrate;
 	int complexity;
 	int signal;
+	bool chaining;
```

Comment:
Make this `const` and use a C++ initializer in the constructor.

Reply:
rest of things are parsed and initialized inside PreparedEncoder::PreparedEncoder()'s body.
I just did things in consistent way. Should I really change this?

Reply:
though they all had reason to be in body, because they couldn't be done with just initializer so I guess I will

Reply:
That's because the other attributes have more complex parsers. But I welcome patches which move the parsers into separate (throwing) functions and make the other attributes `const` as well. The old code was from a time when MPD had C++ exceptions disabled (`-fno-exceptions`) and error handling was different (more complicated).